tasks/static_quality_gates/lib/gates_lib.py

```diff
@@ -0,0 +1,171 @@
+import glob
+import math
+import os
+import types
+from datetime import datetime
+from types import SimpleNamespace
+
+from invoke.exceptions import Exit
+
+from tasks.libs.common.color import color_message
+from tasks.libs.common.constants import ORIGIN_CATEGORY, ORIGIN_PRODUCT, ORIGIN_SERVICE
+from tasks.libs.common.datadog_api import create_gauge, send_metrics
+from tasks.libs.common.utils import get_metric_origin
+
+
+def argument_extractor(entry_args, **kwargs) -> SimpleNamespace:
+    """
+    Allow clean extraction of arguments from parsed quality gates, also allows to execute pre-process function on kwargs
+
+    :param entry_args: Dictionary containing parsed arguments from a static quality gate
+    :param kwargs: Dictionary containing arguments that we want to extract (optionally pre-process function to apply as values)
+    :return: SimpleNamespace with extracted arguments as attributes
+    """
+    for key in kwargs.keys():
+        if isinstance(kwargs[key], types.FunctionType):
+            kwargs[key] = kwargs[key](entry_args[key])
+        else:
+            kwargs[key] = entry_args[key]
+    return SimpleNamespace(**kwargs)
+
+
+def byte_to_string(size):
+    if not size:
+        return "0B"
+    size_name = ("B", "KiB", "MiB", "GiB", "TiB", "PiB", "EiB", "ZiB", "YiB")
+    i = int(math.log(size, 1024))
+    p = math.pow(1024, i)
+    s = round(size / p, 2)
+    return f"{s}{size_name[i]}"
+
+
+def string_to_byte(string):
```

Comment:
🥜 I would probably not use `string`, even if it's not a reserved word in python it could be confusing. You can add the type annotation to make it even clearer (because you convert to `byte` but not to the byte type :) 
```suggestion
def string_to_byte(size :str) -> int:
```

Reply:
Agreed 👍